tests/migration_test.go

```diff
@@ -4194,25 +4136,97 @@ func temporaryTLSConfig() *tls.Config {
 	Eventually(func() bool {
-		node, err = virtClient.CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
+		node, err = virtClient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+
+		value, exists := node.Annotations[v1.LabellerSkipNodeAnnotation]
+		return exists && value == "true"
+	}, 30*time.Second, time.Second).Should(BeTrue(), fmt.Sprintf("node %s is expected to have annotation %s", nodeName, v1.LabellerSkipNodeAnnotation))
+
+	return node
+}
+
+func resumeNodeLabeller(nodeName string, virtClient kubecli.KubevirtClient) *k8sv1.Node {
+	var err error
+	var node *k8sv1.Node
+
+	node, err = virtClient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	if _, isNodeLabellerStopped := node.Annotations[v1.LabellerSkipNodeAnnotation]; !isNodeLabellerStopped {
+		// Nothing left to do
+		return node
+	}
+
+	By(fmt.Sprintf("Patching node to %s not include %s annotation", nodeName, v1.LabellerSkipNodeAnnotation))
+	libnode.RemoveAnnotationFromNode(nodeName, v1.LabellerSkipNodeAnnotation)
+
+	// In order to make sure node-labeller has updated the node, the host-model label (which node-labeller
+	// makes sure always resides on any node) will be removed. After node-labeller is enabled again, the
+	// host model label would be expected to show up again on the node.
+	By(fmt.Sprintf("Removing host model label %s from node %s (so we can later expect it to return)", v1.HostModelCPULabel, nodeName))
+	for _, label := range node.Labels {
+		if strings.HasPrefix(label, v1.HostModelCPULabel) {
+			libnode.RemoveLabelFromNode(nodeName, label)
+		}
+	}
+
+	wakeNodeLabellerUp(virtClient)
+
+	By(fmt.Sprintf("Expecting node %s to not include %s annotation", nodeName, v1.LabellerSkipNodeAnnotation))
+	Eventually(func() error {
+		node, err = virtClient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
 		Expect(err).ShouldNot(HaveOccurred())
 
-		value, ok := node.Annotations[v1.LabellerSkipNodeAnnotation]
-		return ok && value == "true"
-	}, 30*time.Second, time.Second).Should(BeTrue())
+		_, exists := node.Annotations[v1.LabellerSkipNodeAnnotation]
+		if exists {
+			return fmt.Errorf("node %s is expected to not have annotation %s", node.Name, v1.LabellerSkipNodeAnnotation)
+		}
+
+		foundHostModelLabel := false
+		for labelKey, _ := range node.Labels {
+			if strings.HasPrefix(labelKey, v1.HostModelCPULabel) {
+				foundHostModelLabel = true
+				break
+			}
+		}
+		if !foundHostModelLabel {
+			return fmt.Errorf("node %s is expected to have a label with %s prefix. this means node-labeller is not enabled for the node", nodeName, v1.HostModelCPULabel)
+		}
+
+		return nil
+	}, 30*time.Second, time.Second).ShouldNot(HaveOccurred())
 
 	return node
 }
 
+func wakeNodeLabellerUp(virtClient kubecli.KubevirtClient) {
```

Comment:
SuperNit:
```suggestion
func wakeUpNodeLabeller(virtClient kubecli.KubevirtClient) {
```

Reply:
Yes it is, hence the "SuperNit" :) It's just slightly harder to read and slightly frowned upon. (see what I did there? :smile: )